assets/app/data/crowbar/services/upgrade-status.factory.spec.js

```diff
@@ -131,6 +136,77 @@ describe('Upgrade Status Factory', function () {
             should.exist(upgradeStatusFactory);
         });
 
+        describe('syncStatusFlags function', function () {
+            it('should be defined', function () {
+                should.exist(upgradeStatusFactory.syncStatusFlags);
+                expect(upgradeStatusFactory.syncStatusFlags).toEqual(jasmine.any(Function));
+            });
+
+            describe('when received status is completed', function () {
+                beforeEach(function () {
+                    bard.mockService(upgradeFactory, {
+                        getStatus: $q.when(completedUpgradeResponse)
+                    });
+
+                    flagsObject = { running: true, completed: false };
+
+                    upgradeStatusFactory.syncStatusFlags(
+                        testedStep, flagsObject, mockedRunningCallback, mockedCompletedCallback
+                    );
+
+                    $rootScope.$digest();
+                });
+
+                it('should call completed callback', function () {
+                    expect(mockedCompletedCallback).toHaveBeenCalledTimes(1);
+                });
+
+                it('should not call running callback', function () {
+                    expect(mockedRunningCallback).not.toHaveBeenCalledTimes(1);
```

Comment:
If `onRunning` is executed none, 2 or more times, this assertion will be correct. Please fix it